server/sonar-web/src/main/js/api/time-machine.js

```diff
@@ -38,11 +38,39 @@ type Response = {
 export const getTimeMachineData = (
   component: string,
   metrics: Array<string>,
-  other?: {}
+  other?: { p?: number, ps?: number, from?: string, to?: string }
 ): Promise<Response> =>
   getJSON('/api/measures/search_history', {
     component,
     metrics: metrics.join(),
     ps: 1000,
     ...other
   });
+
+export const getAllTimeMachineData = (
+  component: string,
+  metrics: Array<string>,
+  other?: { p?: number, ps?: number, from?: string, to?: string },
+  prev?: Response
+): Promise<Response> =>
+  getTimeMachineData(component, metrics, other).then((r: Response) => {
```

Comment:
because you're loading everything, set the `ps` to the maximum available value (500)